test/hotspot/jtreg/runtime/cds/appcds/dynamicArchive/TestAutoCreateSharedArchiveUpgrade.java

```diff
@@ -61,21 +72,33 @@ public class TestAutoCreateSharedArchiveUpgrade {
     private static String newJVM;
 
     public static void main(String[] args) throws Throwable {
-        setupJVMs();
-        doTest();
+        // Get OS and CPU type
+        String arch = props.getProperty("os.arch");
+        String os = getOsId();
+
+        // Earliest testable version is 19
+        int n = java.lang.Runtime.version().major() - 1;
```

Comment:
Can this just do n-1 and not download every release? So if JDK 20 works with JDK 19, then JDK 21 works with JDK 20 and transitively would work with 19?

Reply:
I think we should start the loop from n-1 and go downwards. After the first iteration:

```
if (System.getProperty("test.autocreatesharedarchive.all.jdk.versions") == null) {
  break;
}
```

This should be enough for day-to-day testing. For thoroughness, we can add a task at tier 7 or 8 to specify `-Dtest.autocreatesharedarchive.all.jdk.versions` and check against all applicable JDK versions.